src/SignalR/clients/csharp/Client.Core/src/Internal/InvocationRequest.cs

```diff
@@ -82,7 +93,15 @@ public override void Complete(CompletionMessage completionMessage)
             if (completionMessage.Result != null)
             {
                 Log.ReceivedUnexpectedComplete(Logger, InvocationId);
+
+                if (TryBeginStopActivity())
+                {
+                    Activity.SetStatus(ActivityStatusCode.Error);
+                    Activity.Stop();
+                }
+
                 _channel.Writer.TryComplete(new InvalidOperationException("Server provided a result in a completion response to a streamed invocation."));
+                return;
```

Comment:
@BrennanConroy Is adding this return correct? It seems like this method should exit if there is no result, and the return was missing.

Reply:
Not returning and calling fail after this will log the error but the user won't see it from the channel reader. Instead, they see the exception reporting that the stream incorrectly returned a response.

Is that what you want?

Reply:
Actually, I think the added return is good. You're not supposed to set both error and result (and the server won't), and like you point out, the channel will already be completed with an InvalidOperationException.